### PR TITLE
Updated DataLake xml comments

### LINKED ISSUE #8347: [Storage][DataLake] Cleanup XML comments for DataLake
Check through XML comments for DataLake and update them if needed. Possible copy paste mistakes.

Reference PR #8235

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/8347

### CODE REVIEW

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeServiceClient.cs [hunk 1]
```diff
@@ -548,13 +548,13 @@ public virtual DataLakeFileSystemClient GetFileSystemClient(string fileSystemNam
         /// <param name="publicAccessType">
         /// Optionally specifies whether data in the file system may be accessed
         /// publicly and the level of access. <see cref="PublicAccessType.FileSystem"/>
-        /// specifies full public read access for file system and blob data.
-        /// Clients can enumerate blobs within the file system via anonymous
+        /// specifies full public read access for file system and path data.
+        /// Clients can enumerate paths within the file system via anonymous
         /// request, but cannot enumerate file systems within the storage
         /// account.  <see cref="PublicAccessType.Path"/> specifies public
```
Comment (jaschrep-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeServiceClient.cs]: Are we okay with stating that these are file systems within the account even though it's just a blob account being accessed a certain way? I don't have an alternative, just calling it out to see if there's a better option.

File: sdk/storage/Azure.Storage.Files.DataLake/src/Models/DataLakeLease.cs [hunk 2]
```diff
@@ -6,22 +6,25 @@
...
     /// <summary>
-    /// Lease
+    /// Lease.
     /// </summary>
     public class DataLakeLease
     {
         /// <summary>
-        /// The ETag contains a value that you can use to perform operations conditionally. If the request version is 2011-08-18 or newer, the ETag value will be in quotes.
+        /// The ETag contains a value that you can use to perform operations conditionally.
+        /// If the request service version is 2011-08-18 or newer, the ETag value will be in quotes.
```
Comment (jaschrep-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/Models/DataLakeLease.cs]: I thought the SDKs stripped quotes so they would all be the same. Is this not the case?

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/Models/DataLakeLease.cs]: Etag.ToString() returns the ETag with quotes.

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeServiceClient.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeServiceClient.cs]: We refer the container as "file system" and the blob as "path/file/directory" through the Data Lake SDK.

**Review by jaschrep-msft — APPROVED** [review]

**Pull request merged**